lib/appscale_tools.py

```diff
@@ -625,14 +631,21 @@ def upload_app(cls, options):
     if app_exists:
       time.sleep(20)  # give the AppController time to restart the app
 
-    serving_host, serving_port = userappclient.get_serving_info(app_id,
-      options.keyname)
-    RemoteHelper.sleep_until_port_is_open(serving_host, serving_port,
-      options.verbose)
+    # Makes a call to the AppController to get all the stats and looks
+    # through them for the http port the app can be reached on.
+    result = acc.get_all_stats()
+    #TODO: Check if any handling of JSON response is needed because on time out,
+    #TODO: it returns an error saying No JSON object can be decoded.
+    json_result = json.loads(result)
+    apps_result = json_result['apps']
+    current_app = apps_result[app_id]
```

Comment:
Yes I need to have a try catch & retry logic in place here in that case where app_id is not present in the stats yet. I will do that!